edb/pgsql/compiler/dml.py

```diff
@@ -1068,7 +1101,31 @@ def process_link_update(
         delqry = None
 
     if shape_op is qlast.ShapeOp.SUBTRACT:
-        return data_cte
+        prev_expr = ir_set.expr
+        ir_set.expr = None
+
+        # Turn the IR of the expression on the right side of :=
```

Comment:
I think this setup needs some explanation because I had to spend a bunch of time puzzling on it and I'm not 100% sure I'm right:
1. We clear the expr on the ir_set so that it just represents a pointer to that element
2. We add an overlay to filter out the things we're deleting
3. We "recompile" the ir_set to compute the rows that will be remaining
4. We fix up the ir_set

Mucking around with the ir_set strikes me as kind of scary, so it definitely should be called out clearly, at least.

Reply:
(Should we only be doing this when we are actually enforcing the constraint?)

Reply:
Ugh, it was a quick hack, and it totally slipped my mind to fix this properly.  Your analysis is correct, and I'll add that as a comment.